recipes/volume.rb

```diff
@@ -94,13 +94,18 @@
   end
 end
 
-# Make sure that there is a 'mysql' directory on the mount point of the volume
 directory new_mysql_dir do
   owner 'mysql'
   group 'mysql'
   action :create
 end
 
+# Make sure that the permissions for the  'mysql' directory are set correctly.
+# When recovering from a backup uids could have changed.
+execute "change permissions #{new_mysql_dir} owner" do
+  command "chown --recursive --silent mysql:mysql #{new_mysql_dir}"
+end
```

Comment:
There should be a guard here so prevent this from running when it does not have to.